### DISCUSSION TITLE
prisma 0 to many relationship

### DISCUSSION THREAD

**Discussion by brahimgb** [prisma/prisma#18661] [Q&A] [answered]
i read the Prisma documentation, but i can`t find 0 to many relationship there is only 1 to many relationship is there's a way to do that ?

**Comment by nurul3101** [comment]
Hello @ibrahimgb 👋 

To make a zero-to-many relation in Prisma, you can make the relation field optional on the related model.

For example: Considering two models, `Writer` and `Book`. A writer can have zero or many books and also a book can exist without any writer, you can define this relation like this:
```prisma
model Writer {
  id    Int    @id @default(autoincrement())
  name  String
  books Book[]
}

model Book {
  id        Int     @id @default(autoincrement())
  title     String
  writerId  Int?
  writer    Writer? @relation(fields: [writerId], references: [id])
}
```

Here the writer relation is marked as optional with (?), which would signify that a book can exist without any writer as well.

**Reply by nurul3101** [reply, marked as answer]
Hello @ibrahimgb 👋 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by brahimgb** [comment]
Will do, thenks for the reminder 🎗️

**Reply by nurul3101** [reply]
No worries! :)
